### PR TITLE
Create a dependencyBuild directly

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

File: pkg/reconciler/dependencybuild/dependencybuild.go [hunk 1]
```diff
@@ -1054,6 +1058,27 @@ func (r *ReconcileDependencyBuild) createRebuiltArtifacts(ctx context.Context, l
...
+func (r *ReconcileDependencyBuild) createArtifacts(ctx context.Context, log logr.Logger, pr *tektonpipeline.PipelineRun, db *v1alpha1.DependencyBuild, deployed []string) error {
+	db.Status.DeployedArtifacts = deployed
+
+	for _, i := range deployed {
+		ab := v1alpha1.ArtifactBuild{}
+		ab.Namespace = pr.Namespace
+		ab.Name = artifactbuild.CreateABRName(i)
+		ab.Annotations = map[string]string{artifactbuild.DependencyAnnotation: db.Name}
+		ab.Spec.GAV = i
+		if err := controllerutil.SetOwnerReference(&ab, db, r.scheme); err != nil {
```
Comment (rnc) [inline pkg/reconciler/dependencybuild/dependencybuild.go:1070]: After further testing this isn't correct I think. It needs to be `controllerutil.SetOwnerReference(db, &ab, r.scheme)` but I don't think we can have it in this function as the caller calls patchStatus.

Comment (rnc) [reply pkg/reconciler/dependencybuild/dependencybuild.go:1070]: I am trying an experimental change to update it elsewhere but currently hitting a mixture of rbac problems and 'object has been modified' ....

File: pkg/reconciler/dependencybuild/dependencybuild.go [hunk 2]
```diff
@@ -788,6 +788,10 @@ func (r *ReconcileDependencyBuild) handleBuildPipelineRunReceived(ctx context.Co
 			if err != nil {
 				return reconcile.Result{}, err
 			}
+			err = r.createArtifacts(ctx, log, pr, db, deployed)
```
Comment (stuartwdouglas) [inline pkg/reconciler/dependencybuild/dependencybuild.go:791]: I don't think we need to create `ArtifactBuild` objects for this case, we already have `RebuiltArtifact` as a marker that the objects were created.

File: java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/artifactwatch/ScmLookup.java [hunk 3]
```diff
@@ -80,27 +82,72 @@ public void onAdd(ArtifactBuild newObj) {
...
-                                scm.setScmURL(result.getRepoInfo().getUri());
-                                scm.setCommitHash(result.getHash());
-                                scm.setPath(result.getRepoInfo().getPath());
-                                scm.set_private(result.getRepoInfo().isPrivateRepo());
-                                scm.setTag(result.getTag());
-                                Log.infof("Adding artifactBuild with GAV %s with URI %s and hash %s ",
-                                        newObj.getSpec().getGav(), result.getRepoInfo().getUri(), result.getHash());
+
+                                // If the DependencyBuild was created directly no need to look up the GAV source, instead gather
+                                // from existing dependency.
```
Comment (stuartwdouglas) [inline java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/artifactwatch/ScmLookup.java:88]: Like I mentioned below I don't think we need to handle this case.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
I think that in addition to the CLI changes we also need a way to rebuild the DependencyBuild objects, at the moment this is handled via annotations on the ArtifactBuild, but if there is not corresponding ArtifactBuild then we can't directly rebuild them.

**Pull request merged**